loopy/kernel/data.py

```diff
@@ -296,24 +344,36 @@ def stringify(cls, val):
 # {{{ arguments
 
 class KernelArgument(ImmutableRecord):
-    """Base class for all argument types"""
+    """Base class for all argument types.
+
+    .. attribute:: name
+    .. attribute:: dtype
+    .. attribute:: is_output
+    .. attribute:: is_input
+
+    .. automethod:: supporting_names
+    """
+    name: str
+    dtype: LoopyType
+    is_output: bool
+    is_input: bool
```

Comment:
Does this also need a `dataclass` decorator?

Reply:
As above. This *should* become a dataclass at some point, but for now too much fun is being had in the constructor.